data-prepper-plugins/armeria-common/src/test/java/org/opensearch/dataprepper/plugins/GrpcBasicAuthenticationProviderTest.java

```diff
@@ -152,12 +154,12 @@ void httpRequest_with_correct_authentication_responds_OK() {
                     .method(HttpMethod.POST)
                     .path("/grpc.health.v1.Health/Check")
                     .contentType(MediaType.JSON_UTF_8)
-                    .build());
+                    .build(),
+                    HttpData.of(Charset.defaultCharset(), "{\"healthCheckConfig\":{\"serviceName\": \"test\"} }"));
 
             final AggregatedHttpResponse httpResponse = client.execute(request).aggregate().join();
 
-            // TODO: Figure out how to get SampleHealthGrpcService to return a status of 200
-            assertThat(httpResponse.status(), equalTo(HttpStatus.SERVICE_UNAVAILABLE));
+            assertThat(httpResponse.status(), equalTo(HttpStatus.OK));
```

Comment:
Do we have a test for bad request ?

Reply:
This is a test for authentication. The test was making a request against a health check endpoint. It didn't care about the response except to be sure that it was not blocked as an unauthenticated request. So the goal is to verify that we can reach a test endpoint.